api/nodes/views.py

```diff
@@ -468,10 +482,28 @@ def get_default_queryset(self):
             contributors.append(contributor)
         return contributors
 
+    def get_serializer_class(self):
+        if self.request.method == 'POST':
+            return NodeContributorsCreateSerializer
+        else:
+            return NodeContributorsSerializer
+
     # overrides ListAPIView
     def get_queryset(self):
         return self.get_queryset_from_request()
 
+    # overrides ListCreateAPIView
+    def get_parser_context(self, http_request):
+        """
+        Tells parser that we are creating a relationship
+        """
+        return {
+            'is_relationship': True,
+            'view': self,
+            'args': getattr(self, 'args', ()),
+            'kwargs': getattr(self, 'kwargs', {})
+        }
```

Comment:
Instead of duplicating logic from the parent method (and ignoring it completely), invoke it as a super call, then add to the result. That way you will benefit from any changes made to the parent method later.

```python
res = super(NodeContributorsList, self).get_parser_context(self, http_request)
res['is_relationship'] = True
return res
```